src/etl/ReportingETL.cpp

```diff
@@ -933,38 +937,55 @@ ReportingETL::loadCacheAsync(uint32_t seq)
     {
         std::optional<ripple::uint256> start = cursors[i];
         std::optional<ripple::uint256> end = cursors[i + 1];
-        std::thread t{[this, seq, start, end, numRemaining]() {
-            std::optional<ripple::uint256> cursor = start;
-            while (true)
-            {
-                auto res = Backend::synchronousAndRetryOnTimeout(
-                    [this, seq, &cursor](auto yield) {
-                        return backend_->fetchLedgerPage(
-                            cursor, seq, 4096, yield);
-                    });
-                backend_->cache().update(res.objects, seq, true);
-                if (!res.cursor || (end && *(res.cursor) > *end))
-                    break;
-                BOOST_LOG_TRIVIAL(debug)
-                    << "Loading cache. cache size = "
-                    << backend_->cache().size()
-                    << " - cursor = " << ripple::strHex(res.cursor.value());
-                cursor = std::move(res.cursor);
-            }
-            if (--(*numRemaining) == 0)
-            {
-                BOOST_LOG_TRIVIAL(info) << "Finished loading cache";
-                backend_->cache().setFull();
-                delete numRemaining;
-            }
-            else
-            {
-                BOOST_LOG_TRIVIAL(info)
-                    << "Finished a cursor. num remaining = " << *numRemaining;
-            }
-        }};
-        t.detach();
+        cacheDownloadStrands_.emplace_back(ioContext_);
+        boost::asio::spawn(
+            cacheDownloadStrands_.back(),
+            [this, seq, start, end, numRemaining](
+                boost::asio::yield_context yield) {
+                std::optional<ripple::uint256> cursor = start;
+                while (true)
+                {
+                    auto res =
+                        Backend::retryOnTimeout([this, seq, &cursor, &yield]() {
+                            return backend_->fetchLedgerPage(
+                                cursor, seq, 256, yield);
```

Comment:
It's sort of arbitrary. When it was 4096, it could take a long time to fetch a single page over a WAN, so I changed it to 256 to see progress more often. I feel that 256 is better though, since 4096 could lead to `4096 * num_cursors` objects being fetched at once. There are usually over 100 cursors.